api/server/server.go

```diff
@@ -790,30 +790,23 @@ func postContainersCreate(eng *engine.Engine, version version.Version, w http.Re
 		return err
 	}
 	var (
-		job          = eng.Job("create", r.Form.Get("name"))
-		outWarnings  []string
-		stdoutBuffer = bytes.NewBuffer(nil)
-		warnings     = bytes.NewBuffer(nil)
+		warnings []string
+		name     = r.Form.Get("name")
+		env      = new(engine.Env)
 	)
 
-	if err := job.DecodeEnv(r.Body); err != nil {
+	if err := env.Decode(r.Body); err != nil {
```

Comment:
I don't think making the env typed should block this PR. I think it should be a separated discussion. It touches more areas than just these two commands and we should be careful.